Use Github Actions 'checkout@v2' instead of v1 to prevent CI re-run errors

##### SUMMARY

Testing the PR https://github.com/ansible-collections/kubernetes/pull/61, I've found errors like:

```
git checkout --progress --force 29436c4030820f6feb760e730d09c2d819ee5a57
##[error]fatal: reference is not a tree: 29436c4030820f6feb760e730d09c2d819ee5a57
##[error]Git checkout failed with exit code: 128
##[error]Exit code 1 returned from process: file name '/home/runner/runners/2.169.0/bin/Runner.PluginHost', arguments 'action "GitHub.Runner.Plugins.Repository.v1_0.CheckoutTask, Runner.Plugins"'.
```

It seems like that was tracked in https://github.com/actions/checkout/issues/23, and that issue recommends upgrading to the checkout action v2 (we're currently using v1). I don't believe this will break anything in our CI workflow, but it should lead to better stability when re-running failed jobs like I was doing for the PR linked earlier.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

CI

With v2 pathing I'm getting:

```
.
└── ansible_collections
    └── community
        └── kubernetes
            ├── CHANGELOG.md
            ├── LICENSE
            ├── README.md
            ├── codecov.yml
            ├── galaxy.yml
            ├── molecule
            │   └── default
            │       ├── converge.yml
            │       ├── molecule.yml
            │       ├── tasks
            │       │   ├── append_hash.yml
            │       │   ├── apply.yml
            │       │   ├── crd.yml
            │       │   ├── delete.yml
            │       │   ├── exec.yml
            │       │   ├── full.yml
            │       │   ├── lists.yml
            │       │   ├── log.yml
            │       │   └── waiter.yml
            │       └── vars
            │           └── main.yml
            ├── plugins
            │   ├── connection
            │   │   └── kubectl.py
            │   ├── doc_fragments
            │   │   ├── __init__.py
            │   │   ├── k8s_auth_options.py
            │   │   ├── k8s_name_options.py
            │   │   ├── k8s_resource_options.py
            │   │   ├── k8s_scale_options.py
            │   │   └── k8s_state_options.py
            │   ├── filter
            │   │   └── k8s.py
            │   ├── inventory
            │   │   ├── k8s.py
            │   │   └── openshift.py
            │   ├── lookup
            │   │   └── k8s.py
            │   ├── module_utils
            │   │   ├── __init__.py
            │   │   ├── common.py
            │   │   ├── raw.py
            │   │   └── scale.py
            │   └── modules
            │       ├── __init__.py
            │       ├── k8s.py
            │       ├── k8s_auth.py
            │       ├── k8s_exec.py
            │       ├── k8s_info.py
            │       ├── k8s_log.py
            │       ├── k8s_scale.py
            │       └── k8s_service.py
            └── tests
                ├── integration
                │   └── targets
                │       └── kubernetes
                │           ├── README.md
                │           ├── aliases
                │           ├── defaults
                │           │   └── main.yml
                │           ├── files
                │           │   ├── crd-resource.yml
                │           │   ├── kuard-extra-property.yml
                │           │   ├── kuard-invalid-type.yml
                │           │   └── setup-crd.yml
                │           ├── handlers
                │           │   └── main.yml
                │           ├── library
                │           │   ├── README.md
                │           │   └── test_tempfile.py
                │           ├── meta
                │           │   └── main.yml
                │           └── tasks
                │               ├── main.yml
                │               ├── older_openshift_fail.yml
                │               ├── openshift.yml
                │               ├── setup_remote_tmp_dir.yml
                │               ├── validate_installed.yml
                │               └── validate_not_installed.yml
                └── sanity
                    ├── ignore-2.10.txt
                    └── ignore-2.9.txt

26 directories, 59 files
ERROR: The current working directory must be at or below:

 - an Ansible collection: {...}/ansible_collections/{namespace}/{collection}/

Current working directory: /home/runner/work/kubernetes/kubernetes
##[error]Process completed with exit code 1.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -17,7 +17,7 @@ jobs:
         python_version: ['3.6']
     steps:
       - name: Check out code
-        uses: actions/checkout@v1
+        uses: actions/checkout@v2
         with:
           path: ansible_collections/community/kubernetes
 
@@ -31,6 +31,7 @@ jobs:
 
       - name: Run sanity tests on Python ${{ matrix.python_version }}
         run: ansible-test sanity --docker -v --color --python ${{ matrix.python_version }}
+        working-directory: ./ansible_collections/community/kubernetes
 
   integration:
     runs-on: ubuntu-latest
@@ -39,7 +40,7 @@ jobs:
         python_version: ['3.6']
     steps:
       - name: Check out code
-        uses: actions/checkout@v1
+        uses: actions/checkout@v2
         with:
           path: ansible_collections/community/kubernetes
 
@@ -53,9 +54,11 @@ jobs:
 
       - name: Run integration tests on Python ${{ matrix.python_version }}
         run: ansible-test integration --docker -v --color --retry-on-error --python ${{ matrix.python_version }} --continue-on-error --diff --coverage
+        working-directory: ./ansible_collections/community/kubernetes
 
       - name: Generate coverage report.
         run: ansible-test coverage xml -v --requirements --group-by command --group-by version
+        working-directory: ./ansible_collections/community/kubernetes
 
       - uses: codecov/codecov-action@v1
         with:
@@ -68,7 +71,7 @@ jobs:
         python_version: ['3.7']
     steps:
       - name: Check out code
-        uses: actions/checkout@v1
+        uses: actions/checkout@v2
         with:
           path: ansible_collections/community/kubernetes
 
@@ -89,7 +92,8 @@ jobs:
       - name: Create default collection path symlink
         run: |
           mkdir -p /home/runner/.ansible
-          ln -s /home/runner/work/kubernetes /home/runner/.ansible/collections
+          ln -s /home/runner/work/kubernetes/kubernetes /home/runner/.ansible/collections
 
       - name: Run molecule default test scenario
         run: molecule test
+        working-directory: ./ansible_collections/community/kubernetes

